### PR TITLE
fix: Fix the inaccurate judgment of ipykernel version in debug feature.

### PR DESCRIPTION
**Pull request by realDuang**
Fixes #https://github.com/microsoft/vscode-jupyter/issues/12801

### CODE REVIEW

File: src/notebooks/debugger/helper.ts [hunk 1]
```diff
@@ -25,10 +25,18 @@ import ipykernel
...
-        const version = output[0].text.toString().split('.');
-        const majorVersion = Number(version[0]);
-        return majorVersion >= 6 ? IpykernelCheckResult.Ok : IpykernelCheckResult.Outdated;
+    const versionRegex = /^\d+\.\d+\.\d+$/;
+
+    // It is necessary to traverse all the output to determine the version of ipykernel, some jupyter servers may return extra status metadata
+    for (const line of output) {
+        const lineText = line.text?.toString();
+        if (lineText && versionRegex.test(lineText)) {
+            const [majorVersion] = lineText.split('.').map(Number);
```
Comment (roblourens) [inline src/notebooks/debugger/helper.ts:34]: Why use a regex then ignore the result and do a split instead?

Comment (roblourens) [reply src/notebooks/debugger/helper.ts:34]: You resolved this but it doesn't look like anything changes

Comment (realDuang) [reply src/notebooks/debugger/helper.ts:34]: > You resolved this but it doesn't look like anything changed

Oh hey Rob, I have responsed in channel 3weeks ago and then click the resolve button last week just want to ping you kindly. but somehow my reply disappear. Sorry to confuse you. I will rewrite it.

Comment (realDuang) [reply src/notebooks/debugger/helper.ts]: Have changed the regex way to pick up the major version.

**Comment by roblourens** [comment]
I don't know why these tests are failing...

**Comment by roblourens** [comment]
This test is actually failing consistently, maybe it's actually related to your changes, can you check that? `Delete temp debugging files`

**Comment by realDuang** [comment]
> This test is actually failing consistently, maybe it's actually related to your changes, can you check that? `Delete temp debugging files`

That's weird. I just ran the unit tests locally and they all seem to pass. Is the test case in question included in script `test:unittests`?

17525129-eb49-4233-9284-f6439b3b7b69

**Comment by roblourens** [comment]
When I run this, there is a newline at the end of the result, so your regex doesn't match. A `.trim()` should do it

**Comment by realDuang** [comment]
> When I run this, there is a newline at the end of the result, so your regex doesn't match. A `.trim()` should do it

Yeash you're right. I reproduced this with a local notebook。 I also found that the output type only can be a stream.

**Review by roblourens — APPROVED** [review]
Thanks!

**Review by lramos15 — APPROVED** [review]

**Pull request merged**